volatility3/framework/symbols/linux/__init__.py

```
@@ -190,9 +190,13 @@ def do_get_path(cls, rdentry, rmnt, dentry, vfsmnt) -> Union[None, str]:
 
             parent = dentry.d_parent
             dname = dentry.d_name.name_as_str()
-            path_reversed.append(dname.strip("/"))
+            dname_stripped = dname.strip("/")
+            if dname_stripped:
+                path_reversed.append(dname_stripped)
             dentry = parent
 
+        if path_reversed == []:
+            return ""
```

Is this the right return value?  If `path_reversed` is empty we presumably want to throw an exception and/or return `None` so that people can tell an error's occurred?

---

Yes there is a catch to this, as it should at least return "/" if we hit the break in the global root if-block. The code made it implicit and I didn't catch that at first.

In fact, the "/" and "//" issue is most likely the result of smear, but in the end I don't think returning "" is a good idea anymore. Here is what I have in mind right now, and I prefer discussing the idea before pushing the code:

```diff
+        smeared = False
         while (
             dentry
             and dentry.is_readable()
@@ -190,10 +191,16 @@ class LinuxUtilities(interfaces.configuration.VersionableInterface):
 
             parent = dentry.d_parent
             dname = dentry.d_name.name_as_str()
+            # empty dentry names are most likely
+            # to be the result of smearing
+            if not dname:
+                smeared = True
             path_reversed.append(dname.strip("/"))
             dentry = parent
 
         path = "/" + "/".join(reversed(path_reversed))
+        if smeared:
+            return f"smeared path: {path}"
         return path
```

I though that keeping those broken entries might still be of interest to the user, but adding a disclaimer before the path like we already do here:

https://github.com/volatilityfoundation/volatility3/blob/51f0a3081d5fad41efb1c09d960fe6e374ab4b2d/volatility3/framework/symbols/linux/__init__.py#L233

to make them aware that we detected something off. This would still output paths like `/my_non_smeard_entry/\xffSMEAR\xff/my_non_smeared_entry`, that can at least help identify the inode number even if a part of the name is broken. Paths are "" or start with an "/", so no risk of confusing "smeared path:" with the actual path?

---

Yeah, ok.  I'm a little concerned when returning data that contains additional information (like the starting tag) because it could be used for confusion by a malicious actor (although the chances are extremely low).  I think if we already have code that prepends with `<wanring> thing` then we should do the same here, so something like `f"<potentially smeared> {path}"`.